### PR TITLE
Add type hints and mypy

### PR DESCRIPTION
**Pull request by mxmeinhold**
Add type hints to most functions, in an effort towards #177. Also added mypy and ci for it. Skipped routes, since I didn't want to wrangle the info dict just now.

### CODE REVIEW

File: packet/models.py [hunk 1]
```diff
@@ -114,65 +116,65 @@ def did_sign(self, username, is_csh):
...
+    active_rtp = cast(bool, Column(Boolean, default=False, nullable=False))
+    three_da = cast(bool, Column(Boolean, default=False, nullable=False))
+    webmaster = cast(bool, Column(Boolean, default=False, nullable=False))
+    c_m = cast(bool, Column(Boolean, default=False, nullable=False))
+    w_m = cast(bool, Column(Boolean, default=False, nullable=False))
+    drink_admin = cast(bool, Column(Boolean, default=False, nullable=False))
+    updated = cast(datetime, Column(DateTime, default=datetime.now, onupdate=datetime.now, nullable=False))
 
-    packet = relationship('Packet', back_populates='upper_signatures')
+    packet = cast(Packet, relationship('Packet', back_populates='upper_signatures'))
```
Comment (devinmatte) [inline packet/models.py:153]: I'm surprised this works

**Review by devinmatte — APPROVED** [review]

File: packet/models.py (hunk 1 above)
Comment (mxmeinhold) [reply packet/models.py:153]: Casting is just a no-op at runtime. It's only for the type checker, and it seems to be unconditional.

Comment (mxmeinhold) [reply packet/models.py:153]: Types on all the imports are a mess, which is why they're ignored completely.

Comment (mxmeinhold) [reply packet/models.py:153]: There is the `sqlalchemy[mypy]` thing, which definitely made it slightly better, but it doesn't quite work with `flask_sqlalchemy`, so this casting mess is what we're left with.

**Review by adamhb123 — APPROVED** [review]
LGTM

**Comment by mxmeinhold** [comment]
Force pushing to squash fixups before merge

**Comment by mxmeinhold** [comment]
So, I messed up the merge and undid it. This actually got squashed to 73e55ac and merged as 2c25b06.

**Pull request merged**